packages/orbit-components/src/Loading/index.tsx

```diff
@@ -77,19 +81,23 @@ const Loader = ({ type, customSize }) => {
 };
 const Loading = ({
   loading = false,
+  asComponent = "div",
   type = TYPE_OPTIONS.PAGE_LOADER,
   text,
   children,
   dataTest,
   customSize,
+  title,
   id,
 }: Props) => {
+  const Element = text ? "div" : asComponent;
```

Comment:
Why this restriction?

Reply:
Because `text` prop is rendered inside a `div` (line 119). In case we have optional `asComponent`, it could be anything - any inline-level component (ie. `span` in ButtonPrimitive component) and in this case, we want to avoid having `div` inside inline-level elements.